### DISCUSSION TITLE
Trying to self host on docker. Don't understand what to put in public URLs in env file

### DISCUSSION THREAD

**Discussion by yourfriendaaron** [hearchco/frontend#684] [General]
I keep getting the error `Failed to create API URL: TypeError: Invalid URL` no matter what I try putting in the .env file variables for PUBLIC_URI, API_URI, PUBLIC_API_URI.

Note I don't have this on any domain yet, just running locally on my home server, so using IP addresses not URLs. Are URLs required?

**Comment by aleksasiriski** [comment]
URLs don't have to be domains, they can be IP addresses. These are required for CORS to work correctly. Let's say that your server has the IP `192.168.1.10`:

```
PUBLIC_URI=http://192.168.1.10:3000
API_URI=http://localhost:8000
PUBLIC_API_URI=http://192.168.1.10:8000
```

You set the PUBLIC_URI to be the one you (user/browser) use to connect to the frontend

You set the API_URI to be the one your server uses to connect to the agent

You set the PUBLIC_API_URI to be the one you (user/browser) use to connect to the agent

**Comment by yourfriendaaron** [comment]
Thanks! Sorry I'm not a dev, just a home server tinkerer so not an expert at this stuff.

That seemed to help, in that after I entered all local IPs it moved on to a new error though:

`502: Failed to fetch: Error: CORS error: No 'Access-Control-Allow-Origin' header is present on the requested resource`

**Comment by aleksasiriski** [comment]
Sorry for not replying, been very busy with another project.

Can you explain me your setup in a bit more detail, are you hosting both the frontend and backend on one server, and if so, how? docker-compose? If so, please send it here (with secrets removed)

**Comment by yourfriendaaron** [comment]
No worries. Thanks for helping me out.

I'm hosting via Docker on Unraid. Using the community app store version of the frontend and the agent, which pulls from here.

I somewhat know what I'm doing Docker-wise since I'm successfully hosting a suite of other services like Plex and the *arr family, so I'm not a complete noob I hope.

Screenshots of the configs as follows...
![image](https://github.com/user-attachments/assets/879a201b-12ab-43a9-9da6-2db0142aed7b)
![image](https://github.com/user-attachments/assets/88f0c243-e685-46c5-b01e-e7efddf30dfc)

**Comment by aleksasiriski** [comment]
Huh, wow, didn't know someone made unraid apps for our project, that's very cool :D I've never used unraid, can you give me any hints as to where can I find the source for this "app" so that I can make sure the docker is setup correctly?

Env vars look good, can you at least access `http://10.0.0.3:8000/healthz` to get `OK` response from agent? Also, check `http://10.0.0.3:3000/healthz` to get `OK` from frontend?

**Comment by yourfriendaaron** [comment]
Thanks!

Here's what the unraid app page looks like:
![image](https://github.com/user-attachments/assets/480c9cca-0574-4c9f-a7ed-8fc7a56f34f2)

The maintainer's app template is here:
https://github.com/nwithan8/unraid_templates/blob/main/templates/hearchco_agent.xml

As far as the health pages, `http://10.0.0.3:8000/healthz` reports OK but `http://10.0.0.3:3000/healthz` reports the 502 CORS error.
